### PR TITLE
add preconditions in seters

### PR DESCRIPTION
**Pull request by darkBasic666**
## Summary 
- Add preconditions in seters
## Trello
https://trello.com/c/zaHeAD9q/30-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: training/src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -47,15 +49,13 @@ public long getId() {
...
-    public void setId(long id) {
-        this.id = id;
-    }
-
     public String getTitle() {
         return title;
     }
 
     public void setTitle(String title) {
+        Preconditions.checkNotNull(title);
```
Comment (zeminlu) [inline training/src/main/java/wolox/training/models/Book.java]: Merge both like this, everywhere:
Preconditions.checkArgument(title != null and !title.isEmpty());

File: training/src/main/java/wolox/training/models/Client.java [hunk 2]
```diff
@@ -48,6 +47,7 @@ public LocalDate getBirthdate() {
     }
 
     public void setBirthdate(LocalDate birthdate) {
+        Preconditions.checkNotNull(birthdate);
```
Comment (zeminlu) [inline training/src/main/java/wolox/training/models/Client.java]: You can directly assign the return of a checkNotNull ;)

**Review by zeminlu — CHANGES_REQUESTED** [review]
Looks good! Check my comments!

File: training/src/main/java/wolox/training/models/Client.java (hunk 2 above)
Comment (darkBasic666) [reply training/src/main/java/wolox/training/models/Client.java]: Asi ? 
Preconditions.checkArgument(title != null and !title.isEmpty()).set(title)

**Review by zeminlu — APPROVED** [review]

**Pull request merged**